components/textfield/index.css

```diff
@@ -13,503 +13,200 @@ governing permissions and limitations under the License.
 
 	/* default icon spacing - invalid */
-	--spectrum-textfield-icon-spacing-inline-start-invalid: var(
-		--spectrum-field-text-to-alert-icon-medium
-	);
-	--spectrum-textfield-icon-spacing-inline-end-invalid: var(
-		--spectrum-field-edge-to-alert-icon-medium
-	);
-	--spectrum-textfield-icon-spacing-inline-end-quiet-invalid: var(
-		--spectrum-field-edge-to-alert-icon-quiet
-	);
-	--spectrum-textfield-icon-spacing-block-invalid: var(
-		--spectrum-field-top-to-alert-icon-medium
-	);
-
-		/* default icon spacing - valid */
-	--spectrum-textfield-icon-spacing-inline-start-valid: var(
-		--spectrum-field-text-to-validation-icon-medium
-	);
-	--spectrum-textfield-icon-spacing-inline-end-valid: var(
-		--spectrum-field-edge-to-validation-icon-medium
-	);
-	--spectrum-textfield-icon-spacing-inline-end-quiet-valid: var(
-		--spectrum-field-edge-to-validation-icon-quiet
-	);
-	--spectrum-textfield-icon-spacing-block-valid: var(
-		--spectrum-field-top-to-validation-icon-medium
-	);
+	--spectrum-textfield-icon-spacing-inline-start-invalid: var(--spectrum-text-to-visual-100);
+	--spectrum-textfield-icon-spacing-inline-end-invalid: var(--spectrum-field-edge-to-alert-icon-medium);
+	--spectrum-textfield-icon-spacing-block-invalid: var(--spectrum-component-top-to-workflow-icon-100);
+
+	/* default icon spacing - valid */
+	--spectrum-textfield-icon-spacing-inline-start-valid: var(--spectrum-text-to-visual-100);
+	--spectrum-textfield-icon-spacing-inline-end-valid: var(--spectrum-field-edge-to-validation-icon-medium);
+	--spectrum-textfield-icon-spacing-block-valid: var(--spectrum-field-top-to-validation-icon-medium);
 
 	/* font styles */
 	--spectrum-textfield-font-family: var(--spectrum-sans-font-family-stack);
 	--spectrum-textfield-font-weight: var(--spectrum-regular-font-weight);
+	--spectrum-textfield-line-height: var(--spectrum-line-height-100);
+	--spectrum-textfield-line-height-cjk: var(--spectrum-cjk-line-height-100);
 	--spectrum-textfield-placeholder-font-size: var(--spectrum-font-size-100);
 
 	/* character count */
-	--spectrum-textfield-character-count-font-family: var(
-		--spectrum-sans-font-family-stack
-	);
-	--spectrum-textfield-character-count-font-weight: var(
-		--spectrum-regular-font-weight
-	);
-	--spectrum-textfield-character-count-font-size: var(--spectrum-font-size-75);
-	--spectrum-textfield-character-count-spacing-inline: var(
-		--spectrum-spacing-200
-	);
-	--spectrum-textfield-character-count-spacing-block: var(
-		--spectrum-component-bottom-to-text-75
-	);
-	--spectrum-textfield-character-count-spacing-inline-side: var(
-		--spectrum-side-label-character-count-to-field
-	);
-	--spectrum-textfield-character-count-spacing-block-side: var(
-		--spectrum-side-label-character-count-top-margin-medium
-	);
+	--spectrum-textfield-character-count-color: var(--spectrum-neutral-subdued-content-color-default);
+	--spectrum-textfield-character-count-font-family: var(--spectrum-sans-font-family-stack);
```

Comment:
Yes! You're right, we did talk about it.

In the tokens, we have this defined for `default-font-family`:
```
--spectrum-default-font-family:var(--spectrum-sans-serif-font-family);
--spectrum-sans-serif-font-family:Adobe Clean;
```

And then for `sans-font-family-stack` we have the following:
```
--spectrum-sans-font-family-stack:adobe-clean, var(--spectrum-sans-serif-font-family), 'Source Sans Pro', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Trebuchet MS', 'Lucida Grande', sans-serif;
--spectrum-sans-serif-font-family:Adobe Clean;
```

What we'd talked about with @jawinn before is that they point to the same font but on the off-chance that for some reason the browser can't find that Adobe Clean font, we have a series of back-up fonts that can be used instead--it feels a lot more robust to have more backup options, and since it should render the same if the Adobe Clean font is found, it felt preferable to use the stack instead of the single font token.

Reply:
Beautiful, thanks for re-explaining that for me. Makes sense, and I'll be sure to use it in the future then 😬